spec.md

```diff
@@ -289,6 +290,32 @@ encapsulated within the `data` attribute.
 * Constraints:
   * OPTIONAL
 
+# Size Limits
+
+In order to increase interoperability, the following size limits apply to any
+CloudEvent:
+
+* The CloudEvent serialized as JSON (minified, i.e. without white-space) MUST
+  NOT exceed a size of 128KB.
+* The CloudEvent MUST NOT have more than 100 top-level attributes.
```

Comment:
This is mainly to account for HTTP header count limits, right? If so, there are multiple problems:
* we should leave some room for normal headers (10-20?) that will be there in addition to CloudEvents and extension attributes. Things like `cookie`, `accept-encoding` etc.
* as detailed in #367, with the current HTTP transport binding this limit is leaky as all "application-property" (or whatever) attributes within Maps would also be flattened to their own header

Also, if specifying limits on such basis as HTTP headers, then it would make just as much sense to separate a <8kb size limit for metadata attributes instead of having a total size limit.

Reply:
You're right, an overall limit for metadata attributes make more sense for HTTP headers. Somehow it slipped my mind that we flatten maps into multiple HTTP headers, thanks a lot for bringing this up!

Besides HTTP, some message queues also limit the amount of non-data attributes they allow. E.g. Google Pub/Sub limits it to 100 attributes, AWS SNS to only 10.

I think compatibility with existing implementations is one concern, the other one is what limits are preferred by producers and consumers. I'd especially like input by those that have experience writing messaging middleware.

Speaking as a producer, a limit like `100 top-level attributes` is (at least conceptually) easier to ensure than `8KB of metadata` if that metadata includes user-generated strings. When I opened up the PR, I included a non-KB limit to open up everyone's mind that there are different types of limits possible. I'm happy to change it, though!